code/__HELPERS/unsorted.dm

```diff
@@ -1,2065 +1,2064 @@
-			else
-				names.Add(name)
-				namecounts[name] = 1
-			pois[name] = A
-
-	return pois
-
-/proc/flash_color(mob_or_client, flash_color="#960000", flash_time=20)
-	var/client/C
-	if(istype(mob_or_client, /mob))
-		var/mob/M = mob_or_client
-		if(M.client)
-			C = M.client
-		else
-			return
-	else if(istype(mob_or_client, /client))
-		C = mob_or_client
-
-	if(!istype(C))
-		return
-
-	C.color = flash_color
-	spawn(0)
-		animate(C, color = initial(C.color), time = flash_time)
-
-#define RANDOM_COLOUR (rgb(rand(0,255),rand(0,255),rand(0,255)))
-
-/proc/make_bit_triplet()
-	var/list/num_sample  = list(1, 2, 3, 4, 5, 6, 7, 8, 9)
-	var/result = 0
-	for(var/i = 0, i < 3, i++)
-		var/num = pick(num_sample)
-		num_sample -= num
-		result += (1 << num)
-	return result
-
-/proc/pixel_shift_dir(var/dir, var/amount_x = 32, var/amount_y = 32) //Returns a list with pixel_shift values that will shift an object's icon one tile in the direction passed.
-	amount_x = min(max(0, amount_x), 32) //No less than 0, no greater than 32.
-	amount_y = min(max(0, amount_x), 32)
-	var/list/shift = list("x" = 0, "y" = 0)
-	switch(dir)
-		if(NORTH)
-			shift["y"] = amount_y
-		if(SOUTH)
-			shift["y"] = -amount_y
-		if(EAST)
-			shift["x"] = amount_x
-		if(WEST)
-			shift["x"] = -amount_x
-		if(NORTHEAST)
-			shift = list("x" = amount_x, "y" = amount_y)
-		if(NORTHWEST)
-			shift = list("x" = -amount_x, "y" = amount_y)
-		if(SOUTHEAST)
-			shift = list("x" = amount_x, "y" = -amount_y)
-		if(SOUTHWEST)
-			shift = list("x" = -amount_x, "y" = -amount_y)
-
-	return shift
-
-//Return a list of atoms in a location of a given type. Can be refined to look for pixel-shift.
-/proc/get_atoms_of_type(var/atom/here, var/type, var/check_shift, var/shift_x = 0, var/shift_y = 0)
-	. = list()
-	if(here)
-		for(var/atom/thing in here)
-			if(istype(thing, type) && (check_shift && thing.pixel_x == shift_x && thing.pixel_y == shift_y))
-				. += thing
-
-//gives us the stack trace from CRASH() without ending the current proc.
-/proc/stack_trace(msg)
-	CRASH(msg)
-
-/datum/proc/stack_trace(msg)
-	CRASH(msg)
-
-/proc/pass()
-	return
-
-/atom/proc/Shake(pixelshiftx = 15, pixelshifty = 15, duration = 250)
-	var/initialpixelx = pixel_x
-	var/initialpixely = pixel_y
-	var/shiftx = rand(-pixelshiftx,pixelshiftx)
-	var/shifty = rand(-pixelshifty,pixelshifty)
-	animate(src, pixel_x = pixel_x + shiftx, pixel_y = pixel_y + shifty, time = 0.2, loop = duration)
-	pixel_x = initialpixelx
-	pixel_y = initialpixely
-
-/proc/params2turf(scr_loc, turf/origin, client/C)
-	if(!scr_loc)
-		return null
-	var/tX = splittext(scr_loc, ",")
-	var/tY = splittext(tX[2], ":")
-	var/tZ = origin.z
-	tY = tY[1]
-	tX = splittext(tX[1], ":")
-	tX = tX[1]
-	var/list/actual_view = getviewsize(C ? C.view : world.view)
-	tX = Clamp(origin.x + text2num(tX) - round(actual_view[1] / 2) - 1, 1, world.maxx)
-	tY = Clamp(origin.y + text2num(tY) - round(actual_view[2] / 2) - 1, 1, world.maxy)
-	return locate(tX, tY, tZ)
+/*

```

Comment:
Ok, after having seen a few times what this does, I'm not going to tolerate this anymore.

Either you need to resolve what causes the entire file to be rewritten, or resubmit this entire PR.

As demonstrated by @farie82's recent PR, when it happened there and it got merged, it also showed up in the history, whereby determining blame or who changed what, when, was significantly harder to drill down on; it was just a clothing file, so no big deal.

This has a lot of procs that we really should be able to easily determine the history of.

Reply:
I think its probaly honestly from useing tortisegits merge conflict fixing tool. I do apologize it did this. I actually did not come back to look to see that it freaking rewrote this like that..jesus. I will resubmit later. I am pretty sure it had to do when i was fixing conflicts and i normally don't think to check cause i trust it too much.

Reply:
I have to go to work now though. Sorry fox!